presto-main/src/main/java/com/facebook/presto/sql/planner/iterative/rule/PickTableLayout.java

```diff
@@ -239,6 +273,66 @@ public Result apply(TableScanNode tableScanNode, Captures captures, Context cont
         }
     }
 
+    private static final class TranslateVariableNamesVisitor
```

Comment:
Actually @hellium01 has a nice util to do the same thing (https://github.com/prestodb/presto/pull/12710/commits/21af13229d4491945a8a7061dd980b3252e5915e) with test coverage. Could you check that out? I have asked him to separate that out as a single PR (#12881)

Reply:
It might change in the near future as @rongrong is replacing all `Symbol` with `VariableExpression`.

Reply:
You have `TypeProvider` here, so you can actually construct a proper `VariableReferenceExpressionInliner` It's actually not safe to use the reference variable's type as the inlined variable's type (I've seen test failures related to this). So the proper way to do it is to use `TypeProvider` to construct `VariableReferenceExpression` and use the variable mapping as context.

Reply:
I think what you actually need is something like `RowExpressionNodeInliner` to replace `VariableReferenceExpression` with `ColumnReferenceExpression` back/forth?

Reply:
@rongrong @highker @hellium01 Yi, I found `RowExpressionNodeInliner`. It allows one to replace one RowExpression with another. I could use it to replace VariableReferenceExpression with VariableReferenceExpression, but I don't see ColumnReferenceExpression. Are you suggesting to introduce one?

Reply:
I thought we already introduced `ColumnReferenceExpression` that uses ColumnHandle. If not, can you add support for that? It is going to be very useful for pushdown cases.